lib/dom.js

```diff
@@ -4,6 +4,116 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
+/* Crypto */
+
+// This is a helper type to simplify the specification, it isn't an interface
+// and there are no objects implementing it.
+// https://developer.mozilla.org/en-US/docs/Web/API/BufferSource
+type BufferSource = ArrayBuffer | $ArrayBufferView;
+
+type DOMString = string;
+
+declare interface Crypto extends RandomSource {
+  +subtle: SubtleCrypto;
+}
+
+type CryptoKey$Type =
+  | 'secret'
+  | 'public'
+  | 'private';
+type CryptoKey$Usages =
+  | 'encrypt'
+  | 'decrypt'
+  | 'sign'
+  | 'verify'
+  | 'deriveKey'
+  | 'deriveBits'
+  | 'wrapKey'
+  | 'unwrapKey';
+declare interface CryptoKey {
+  +algorithm: Object;
+  +extractable: Boolean;
+  +type: CryptoKey$Type;
+  +usages: CryptoKey$Usages[];
+}
+
+// TODO: CryptoKeyPair
+declare interface CryptoKeyPair {
+}
+
+// TODO: JSONWebKey
+type JSONWebKey = {};
+
+type RandomSource = {
+    getRandomValues(typedArray: $TypedArray): $TypedArray;
```

Comment:
This should be a generic, because it is always guaranteed to return the same member of $TypedArray, since it just returns the same instance.

Reply:
Is there any reason `getRandomValues<T: $TypedArray>(typedArray: T): T;` didn't work?